validation/validate.py

```diff
@@ -0,0 +1,268 @@
+import json
+import math
+import docker
+import numpy as np
+from argparse import ArgumentParser, Namespace
+from sklearn.metrics import accuracy_score, recall_score, precision_score
+from sklearn.metrics.pairwise import paired_distances
+from sklearn.model_selection import KFold
+from os.path import join, exists
+from enum import Enum
+from typing import List, Union, Tuple, cast
+
+Match = Tuple[str, int, int]
+Mismatch = Tuple[str, int, str, int]
+Pair = Union[Match, Mismatch]
+
+Path = Tuple[str, str]
+Label = bool
+
+
+class DistanceMetric(Enum):
+    COSINE = 'cosine'
+    EUCLIDEAN = 'euclidean'
+
+
+class DistanceMetricException(Exception):
+    pass
+
+
+def evaluate(embeddings: np.ndarray,
+             labels: np.ndarray,
+             num_folds: int,
+             distance_metric: DistanceMetric,
+             subtract_mean: bool,
+             divide_stddev: bool,
+             threshold_start: float,
+             threshold_end: float,
+             threshold_step: float) -> Tuple[np.float, np.float, np.float]:
+    thresholds = np.arange(threshold_start, threshold_end, threshold_step)
+    embeddings1 = embeddings[0::2]
+    embeddings2 = embeddings[1::2]
+    accuracy, recall, precision = _score_k_fold(thresholds,
+                                                embeddings1,
+                                                embeddings2,
+                                                labels,
+                                                num_folds,
+                                                distance_metric,
+                                                subtract_mean,
+                                                divide_stddev)
+    return np.mean(accuracy), np.mean(recall), np.mean(precision)
+
+
+def _read_pairs(pairs_filename: str) -> List[Pair]:
+    pairs = []
+    with open(pairs_filename, 'r') as pair_file:
+        for line_num, line in enumerate(pair_file):
+            if line_num != 0:
```

Comment:
Instead of doing this check all the time, you can just skip the first line:

```py
with open(...) as fobj:
  next(fobj)  # skip initial header line
  for line in fobj:
    ...
```

Reply:
That's a really good idea!